### PR TITLE
Remove stripe-connect prefix and add onboarding

### PR DESCRIPTION
**Pull request by jojo-stripe**

### CODE REVIEW

File: src/shared.ts [hunk 1]
```diff
@@ -1,8 +1,23 @@
...
 export type LoadConnect = () => Promise<StripeConnectWrapper>;
 
+const componentNameMapping: Record<
+  ConnectElementTagName,
+  ConnectElementHTMLName
+> = {
+  payments: "stripe-connect-payments",
+  payouts: "stripe-connect-payouts",
+  "payment-details": "stripe-connect-payment-details",
+  onboarding: "stripe-connect-account-onboarding"
```
Comment (slye-stripe) [inline src/shared.ts:18]: This should be `account-onboarding`

File: types/shared.d.ts [hunk 2]
```diff
@@ -148,9 +148,14 @@ export interface StripeConnectInstance {
...
-
-export type ConnectElementTagName = "payments" | "payouts" | "payment-details";
+  | "stripe-connect-payment-details"
+  | "stripe-connect-account-onboarding";
+
+export type ConnectElementTagName =
+  | "payments"
+  | "payouts"
+  | "payment-details"
+  | "onboarding";
```
Comment (slye-stripe) [inline types/shared.d.ts:158]: This too `account-onboarding`

File: src/shared.ts [hunk 3]
```diff
@@ -102,6 +117,14 @@ const createWrapper = (stripeConnect: any) => {
           }
         }
       });
+      const oldCreate = stripeConnectInstance.create.bind(
```
Comment (slye-stripe) [inline src/shared.ts:120]: What is the purpose of this?

File: examples/rollup/requirements.txt [hunk 4]
```diff
@@ -1,12 +1,12 @@
 certifi==2022.12.7
```
Comment (jorgea-stripe) [inline examples/rollup/requirements.txt:1]: Is this related?

File: types/shared.d.ts [hunk 5]
```diff
@@ -148,9 +148,14 @@ export interface StripeConnectInstance {
 export type ConnectElementHTMLName =
```
Comment (jorgea-stripe) [inline types/shared.d.ts:148]: We don't need to export this type right?

Comment (jorgea-stripe) [reply types/shared.d.ts:148]: We can move it to shared.ts and not export it

File: src/shared.ts (hunk 3 above)
Comment (jojo-stripe) [reply src/shared.ts:120]: `create` on ConnectInstance only takes in HTML names, we basically want to wrap that method so we can pass in the simplified tag name instead. However if we don't bind the method we run into a recursive error since we're basically replacing the create method and calling it at the same time if we only do something like 

`stripeConnectInstance.create = (tagName: ConnectElementTagName) => {
        const htmlName = componentNameMapping[tagName];
        stripeConnectInstance.create(htmlName);
      };`

Comment (jorgea-stripe) [inline src/shared.ts:120]:
```suggestion
      // We wrap create so we can map its different strings to supported components
      const oldCreate = stripeConnectInstance.create.bind(
```

File: src/shared.ts [hunk 6]
```diff
@@ -102,6 +117,14 @@ const createWrapper = (stripeConnect: any) => {
...
+        stripeConnectInstance
+      );
+      stripeConnectInstance.create = (tagName: ConnectElementTagName) => {
+        const htmlName = componentNameMapping[tagName];
+        return oldCreate(htmlName);
```
Comment (jorgea-stripe) [inline src/shared.ts:125]: Should we add this:
```suggestion
        if (!htmlName) {
          throw new Error(`The element ${tagName} is not supported. Please specify one of these ${htmlTagNames.join(', ')}`);
        }
        return oldCreate(htmlName);
```

**Review by jorgea-stripe — CHANGES_REQUESTED** [review]

File: src/shared.ts (hunk 6 above)
Comment (jorgea-stripe) [reply src/shared.ts:125]: Actually - disregard my suggestion. It should be something like this instead:

```
        if (!htmlName) {
          htmlName = tagName;
        }
```

That way we can still specify private components... this is making me think this should perhaps go in `submerchant-surfaces`, however the typescript types consistency argument still applies, so I'd say this is ok!

**Review by jorgea-stripe — CHANGES_REQUESTED** [review]

File: examples/rollup/requirements.txt (hunk 4 above)
Comment (jojo-stripe) [reply examples/rollup/requirements.txt:1]: These version bumps are required for the rollup example to compile

**Review by jorgea-stripe — APPROVED** [review]
👍
